common/workunit/workunit.cpp

```diff
@@ -3027,6 +3028,7 @@ class CWorkUnitFactory : public CInterface, implements IWorkUnitFactory, impleme
                 sortElements(iter, sortOrder.get(), NULL, NULL, unknownAttributes, elements);
                 return conn.getClear();
             }
+            virtual unsigned getTotalElements() { throw MakeStringException(-1, "not implemented"); }
```

Comment:
By the time IElementsPager::getElements() is called, it knows the total.
In the case of the dfu file pager it has a special mechanism to return the total.

But here, it is matching + sorting/filtering all anyway.
The total you want in this and other cases is the the total possible number of matches (after filtering) - which in this implementation is simply elements.ordinality() - which you can tuck away when CQuerySetQueriesPager::getElements() is called.